# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: 8e9fb56cc22a
# Full Hash: 8e9fb56cc22a4a42925ce6dfe0abb6021a5b5afe
# Author: Birunthan Mohanathas <birunthan@mohanathas.com>
# Date: 2015-08-23 21:18:49
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Part 4: Allow current key to be provided with PBackgroundIDBCursor::Continue. r=khuey
#   
#   The key is assumed to be unset for now.
# ==============================================================================

diff -r 1f31c24e2ca0 -r 8e9fb56cc22a dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Wed Aug 19 14:59:25 2015 -0700
+++ b/dom/indexedDB/ActorsParent.cpp	Wed Aug 19 14:59:28 2015 -0700
@@ -7613,7 +7613,7 @@
   RecvDeleteMe() override;
 
   virtual bool
-  RecvContinue(const CursorRequestParams& aParams) override;
+  RecvContinue(const CursorRequestParams& aParams, const Key& key) override;
 };
 
 class Cursor::CursorOpBase
@@ -7699,12 +7699,16 @@
   friend class Cursor;
 
   const CursorRequestParams mParams;
+  const Key mKey;
 
 private:
   // Only created by Cursor.
-  ContinueOp(Cursor* aCursor, const CursorRequestParams& aParams)
+  ContinueOp(Cursor* aCursor,
+             const CursorRequestParams& aParams,
+             const Key& aKey)
     : CursorOpBase(aCursor)
     , mParams(aParams)
+    , mKey(aKey)
   {
     MOZ_ASSERT(aParams.type() != CursorRequestParams::T__None);
   }
@@ -14755,7 +14759,7 @@
 }
 
 bool
-Cursor::RecvContinue(const CursorRequestParams& aParams)
+Cursor::RecvContinue(const CursorRequestParams& aParams, const Key& aKey)
 {
   AssertIsOnBackgroundThread();
   MOZ_ASSERT(aParams.type() != CursorRequestParams::T__None);
@@ -14774,6 +14778,8 @@
 #endif
     ;
 
+  MOZ_ASSERT(aKey.IsUnset());
+
   if (!trustParams && !VerifyRequestParams(aParams)) {
     ASSERT_UNLESS_FUZZING();
     return false;
@@ -14793,7 +14799,7 @@
     return true;
   }
 
-  nsRefPtr<ContinueOp> continueOp = new ContinueOp(this, aParams);
+  nsRefPtr<ContinueOp> continueOp = new ContinueOp(this, aParams, aKey);
   if (NS_WARN_IF(!continueOp->Init(mTransaction))) {
     continueOp->Cleanup();
     return false;